examples/src/main/kotlin/com/samples/jariko/RpgSamples.kt

```diff
@@ -68,4 +70,58 @@ fun execWithCallBack(programSource: String, programArgs: List<String>, jarikoCal
     val commandLineProgram = getProgram(programSource, systemInterface)
     commandLineProgram.singleCall(programArgs, configuration = configuration)
     println("... done.")
-}
\ No newline at end of file
+}
+
+/**
+ * Draft version of how to pass a DS to Jariko.
+ * */
+fun passDSToJariko() {
+    // testing program
+    val pgm = "     DMsg              S              3\n" +
+            "     DP1               DS\n" +
+            "     D Name                           5A\n" +
+            "     D Surname                        5A\n" +
+            "     D Nbr                            5  2\n" +
+            "     C     *ENTRY        PLIST\n" +
+            "     C                   PARM                    P1\n" +
+            "     C     Name          DSPLY\n" +
+            "     C     Surname       DSPLY\n" +
+            "     C     Nbr           DSPLY\n" +
+            "     C     Msg           DSPLY\n" +
+            "     C                   SETON                                        LR"
+    // set len of DS
+    val dsLen = 15
+    val commandLineProgram = getProgram(nameOrSource = pgm)
+    // create instance
+    val dataStructValue = DataStructValue("".padStart(dsLen))
+    // set DS fields fields
+    // note explicitEndOffset is required but not significant
+    dataStructValue.set(
+        field = FieldDefinition(
+            name = "Name",
+            type = StringType(5, false),
+            explicitStartOffset = 0,
+            explicitEndOffset = 0
+        ),
+        value = StringValue("John")
+    )
+    dataStructValue.set(
+        field = FieldDefinition(
+            name = "Surname",
+            type = StringType(5, false),
+            explicitStartOffset = 5,
+            explicitEndOffset = 0
+        ),
+        value = StringValue("Smith")
+    )
+    // note rpgType = RpgType.ZONED is always required regardless of number type definition in rpg program
```

Comment:
Why should I use RpgType.ZONED for all kind of numeric type?

Reply:
This is a very long story. I will try to explain to you.
Jariko only accepts a list of strings as input parameters.
First of all I had to find a way to represent a DS as a string.
I found that DataStructValue (the DataStruct object model in Jariko) is able to provide a good string representation of itself, via the DataStructValue.value function, only for numeric fields defined as RpgType.ZONED.
However we are only talking about a draft version, and these problems will be addressed in a deliverable, in fact this is not the solution but a possible solution.

Reply:
ok, thanks